### PR TITLE
emit port cli version, variants as separate args

### PR DESCRIPTION
**Pull request by aphor**
### What does this PR do?
Changes macports pkg module behavior to conform to macports documentation for single pkg with optional version and variants arguments.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/45413

### Previous Behavior
The module produces a command with a single port name and version (if given), and variants (if given) concatenated as a single positional arg to the "port install" command.

This is incorrect. The command should give version and variants as separate args eg. 
`port install portname @version +variant`
and the current rev, where version kwarg is omitted and variant is given will try to execute\
`port install portname+variant`
which is incorrect.

### New Behavior
give version and variants as separate args eg. 
`port install portname @version +variant`

### Tests written?

No

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by KaiSzuttor** [comment]
@aphor have you checked your PR? I just tried and an exception occurred:
```
An exception occurred in this state: Traceback (most recent call last):
                File "xxx/git/salt/salt/state.py", line 1964, in call
                  **cdata['kwargs'])
                File "xxx/git/salt/salt/loader.py", line 1822, in wrapper
                  return f(*args, **kwargs)
                File "xxx/git/salt/salt/states/pkg.py", line 1663, in installed
                  **kwargs)
                File "xxx/git/salt/salt/modules/mac_ports.py", line 330, in install
                  if 'version' in pparams:
              TypeError: argument of type 'NoneType' is not iterable
```

**Comment by aphor** [comment]
@KaiSzuttor can you provide an example from which I can add a test?

I checked the code with 4 cases, a single port (from the bug example) with all combinations of version and variant given or not.

If you're getting an exception, I think it would be good to first add a test and then fix it.

File: salt/modules/mac_ports.py [hunk 1]
```diff
@@ -323,7 +323,13 @@ def install(name=None, refresh=False, pkgs=None, **kwargs):
 
     formulas_array = []
     for pname, pparams in six.iteritems(pkg_params):
-        formulas_array.append(pname + (pparams or ''))
+        formulas_array.append(pname)
+
+        if 'version' in pparams:
```
Comment (gtmanfred) [inline salt/modules/mac_ports.py]: you should do a `if pparams and 'version' in pparams:`

because it is possible that pparams is a None value.

**Review by gtmanfred — CHANGES_REQUESTED** [review]

**Comment by aphor** [comment]
I think @gtmanfred is probably right, and I'll fix the PR with his suggestion. I'll have to look into setting up a test for the state module which would cover your case @KaiSzuttor.

**Comment by aphor** [comment]
@gtmanfred I dropped both formulas_array.append() calls into an if block testing for True-ish-ness of pparams. It wasn't exactly the changes you requested, but I think it covers your request, and probably fixes the error @KaiSzuttor got.

**Comment by KaiSzuttor** [comment]
Just checked, variant is still ignored for the state file.

**Comment by gtmanfred** [comment]
@aphor did you see @KaiSzuttor last comment about this causing 'variant' to be ignored when passed from a state?

**Comment by aphor** [comment]
@gtmanfred Yeah, I saw that. I've been sidetracked, but I'll be able to deal with this over the weekend because we're going to get snowed in here in Chicago.

**Comment by gtmanfred** [comment]
cool, thanks!

**Comment by aphor** [comment]
@KaiSzuttor @gtmanfred 

The pkg state module's pkg.installed() doesn't support a variant argument, so it makes calls to the mac_ports execution module to install packages without the (optional) variant kwarg.

I looked at adding support for this into the state module, but I think that should be a separate ticket/PR.

I submitted #45967 "pkg state module does not allow specifying variants for macports packages" to address this.

**Review by gtmanfred — APPROVED** [review]

**Pull request merged**